streamz/core.py

```diff
@@ -927,6 +941,14 @@ def __init__(self, *upstreams, **kwargs):
 
         Stream.__init__(self, upstreams=upstreams2, **kwargs)
 
+    def _add_upstream(self, upstream):
+        self.buffers[upstream] = deque()
```

Comment:
So this is the meat of the change here, that instead of an upstream, we have a set of buffers for each upstream. Perhaps a comment in this method override would be useful explanation.